storage/leveldb/leveldb.go

```diff
@@ -57,32 +59,34 @@ func NewDB(path string, batchDelaySeconds int, maxBatchSize int, maxOpenFiles in
 	}
 
 	bldb := &baseLevelDb{
-		db: db,
+		db:   db,
+		path: path,
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
 	dbStore := &DB{
 		baseLevelDb:       bldb,
-		path:              path,
 		maxBatchSize:      maxBatchSize,
 		batchDelaySeconds: batchDelaySeconds,
 		sizeBatch:         0,
-		dbClosed:          make(chan struct{}),
+		cancel:            cancel,
 	}
 
 	dbStore.batch = dbStore.createBatch()
 
-	go dbStore.batchTimeoutHandle()
+	go dbStore.batchTimeoutHandle(ctx)
 
 	runtime.SetFinalizer(dbStore, func(db *DB) {
 		_ = db.Close()
 	})
 
-	log.Debug("opened level db persister", "path", path)
+	crtCounter := atomic.AddUint32(&loggingDBCounter, 1)
+	log.Debug("opened level db persister", "path", path, "created pointer", fmt.Sprintf("%p", bldb.db), "global db counter", crtCounter)
 
 	return dbStore, nil
 }
 
-func (s *DB) batchTimeoutHandle() {
+func (s *DB) batchTimeoutHandle(ctx context.Context) {
 	for {
 		select {
 		case <-time.After(time.Duration(s.batchDelaySeconds) * time.Second):
```

Comment:
use ticker instead of time after?

Reply:
done in both implementations :1st_place_medal: